app/controllers/application_controller.rb

```diff
@@ -31,4 +32,18 @@ def collect_metrics
     duration = Time.zone.now - start
     Rails.logger.info "#{controller_name}##{action_name}: #{duration}s"
   end
+
+  private
+
+  def set_locale
+    I18n.locale = locale
+  end
+
+  def locale
+    LocaleExtractor.new(accept_lang_header).extract
+  end
+
+  def accept_lang_header
+    request.headers['Accept-Language'].presence || ''
```

Comment:
I am not sure if `request.headers['Accept-Language']` will ever be `nil`. I am just being defensive here.